src/Appwrite/Transformation/Transformation.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Appwrite\Transformation;
+
+class Transformation
+{
+    public function __construct(protected Adapter $adapter)
+    {
+    }
+
+    /**
+     * @param array<mixed> $traits
+     */
+    public function isValid(array $traits): bool
```

Comment:
This should be the protected method who always triggers before `return $this->adapter->transform();`